Composer/packages/client/src/recoilModel/dispatchers/trigger.ts

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+/* eslint-disable react-hooks/rules-of-hooks */
+import { useRecoilCallback, CallbackInterface } from 'recoil';
+import { BaseSchema, deleteActions, ITriggerCondition, LgTemplate, LgTemplateSamples, SDKKinds } from '@bfc/shared';
+import get from 'lodash/get';
+
+import { lgFilesState, luFilesState, schemasState, dialogState, localeState } from '../atoms/botState';
+import { dispatcherState } from '../DispatcherWrapper';
+import { dialogsSelectorFamily } from '../selectors';
+import {
+  onChooseIntentKey,
+  generateNewDialog,
+  intentTypeKey,
+  qnaMatcherKey,
+  TriggerFormData,
+} from '../../utils/dialogUtil';
+
+import { setError } from './shared';
+
+const defaultQnATriggerData = {
+  $kind: qnaMatcherKey,
+  errors: { $kind: '', intent: '', event: '', triggerPhrases: '', regEx: '', activity: '' },
+  event: '',
+  intent: '',
+  regEx: '',
+  triggerPhrases: '',
+};
+
+const getDesignerIdFromDialogPath = (dialog, path) => {
+  const value = get(dialog, path, '');
+  const startIndex = value.lastIndexOf('_');
+  const endIndex = value.indexOf('()');
```

Comment:
do we need error checks here if endIndex is -1?

Reply:
Sure, I think we can throw error here if designerId not found.

Reply:
@liweitian can you help confirm this, I'm not 100% sure.

Reply:
should add an error check there.